### PR TITLE
Fix page jumping after codepen loads

### LINKED ISSUE #3017: Loading of embedded examples moves the page when loading specific section
Follow up to #3015

When navigating to specific section of the docs (for example https://vanillaframework.io/docs/patterns/navigation#side-navigation) on slower connection the CodePens loading later make page move and given section is no longer on top of the screen.

**Expected behavior**

When navigating to specific section it should be scrolled to regardless of CodePen loading.

We could make the wrapper div around CodePen reserve the space before the CodePen loads.

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Add a fix scroll function to ensure the codepen loading doesn't move the title out of view

Fixes #3017 

## QA

- Open [demo](https://vanilla-framework-4432.demos.haus/docs/patterns/navigation#side-navigation)
- Check that the "Side navigation" title is in view. 
- Compare to [main](https://vanillaframework.io/docs/patterns/navigation#side-navigation)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.

### CODE REVIEW

File: templates/static/js/example.js [hunk 1]
```diff
@@ -270,6 +270,20 @@
     return scripts;
   }
 
+  function fixScroll() {
+    const titleId = window.location.hash;
+    if (titleId) {
+      const title = document.querySelector(`${titleId}`);
+      var rect = title.getBoundingClientRect();
```
Comment (bartaz) [inline templates/static/js/example.js:277]: This doesn't seem needed?

Comment (bethcollins92) [reply templates/static/js/example.js:277]: Oh no it isn't - I was trying different techniques using scrollTo or scrollIntoView. It still seems flakey though! What's the demo looking like to you?

File: templates/static/js/example.js [hunk 2]
```diff
@@ -293,4 +307,5 @@
       }
     });
   }
+  fixScroll();
```
Comment (bartaz) [inline templates/static/js/example.js:310]: Around line 160+ we have a section that does resizing of the iframes after they load.
I guess this fixScroll() could be called in there? So it triggers after iframe loads, not when page loads.

Comment (bethcollins92) [reply templates/static/js/example.js]: Thanks - that seems to work a treat

File: templates/static/js/example.js [hunk 3]
```diff
@@ -270,6 +271,14 @@
     return scripts;
   }
 
+  function fixScroll() {
+    const titleId = window.location.hash;
+    if (titleId) {
+      const title = document.querySelector(`${titleId}`);
```
Comment (andesol) [inline templates/static/js/example.js:277]: `titleId` seems to be a string, so do we need the backticks?

**Review by andesol — APPROVED** [review]
Such an elegant solution :100: 

Just a minor detail...

File: templates/static/js/example.js (hunk 3 above)
Comment (bethcollins92) [reply templates/static/js/example.js]: Ahh - good spot! I was trying a few diff solutions and forgot to change this back. Thanks!

**Pull request merged**